### PR TITLE
[#IOPID-2400] ioweb fe storage account

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-2400] ioweb fe storage account](https://github.com/pagopa/io-auth-n-identity-domain/commit/33083fed0472481f666f95e6b040270a9a58aabe)

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/prod/storage.tf [hunk 1]
```diff
@@ -2,3 +2,67 @@ data "azurerm_storage_account" "immutable_lv_audit_logs_storage" {
...
+  }
+  access_tier = "Hot"
+
+  resource_group_name                  = azurerm_resource_group.io_web_profile_fe_rg.name
+  subnet_pep_id                        = data.azurerm_subnet.private_endpoints_subnet.id
+  private_dns_zone_resource_group_name = data.azurerm_resource_group.rg_common.name
+
+  # storage should be accessible by CDN via private endpoint
+  # see https://learn.microsoft.com/en-us/azure/frontdoor/standard-premium/how-to-enable-private-link-storage-account
+  force_public_network_access_enabled = false
```
Comment (gquadrati) [inline infra/resources/prod/storage.tf:37]: 👍

File: infra/resources/prod/storage.tf [hunk 2]
```diff
@@ -2,3 +2,67 @@ data "azurerm_storage_account" "immutable_lv_audit_logs_storage" {
...
+  subservices_enabled = {
+    blob = true
+    # not used
+    file  = false
+    queue = false
+    table = false
```
Comment (gquadrati) [inline infra/resources/prod/storage.tf:43]: We can safely remove this settings, as `false` is the default value.

File: infra/resources/prod/storage.tf [hunk 3]
```diff
@@ -2,3 +2,67 @@ data "azurerm_storage_account" "immutable_lv_audit_logs_storage" {
...
+  }
+  blob_features = {
+    versioning = true
+    change_feed = {
+      enabled = false
+    }
+    immutability_policy = {
+      enabled = false
+    }
```
Comment (gquadrati) [inline infra/resources/prod/storage.tf:52]: We can safely remove this settings, as `false` is the default value.

File: infra/resources/prod/storage.tf [hunk 4]
```diff
@@ -2,3 +2,67 @@ data "azurerm_storage_account" "immutable_lv_audit_logs_storage" {
...
+  static_website = {
+    index_document     = "index.html"
+    error_404_document = "it/404/index.html"
+  }
+
+  # TODO: complete this block after the domain name has been picked
+  #custom_domain = {
+  #  name = ""
+  #  use_subdomain = true
+  #}
```
Comment (gquadrati) [inline infra/resources/prod/storage.tf:64]: Do we need it? We have no custom domain specified in the production storage

**Review by gquadrati — CHANGES_REQUESTED** [review]
Even if I like the idea to move everything within the monorepo, I'd rather leave these resources on `io-infra` right now, and then migrate everything all at once.

File: infra/resources/prod/storage.tf (hunk 4 above)
Comment (arcogabbo) [reply infra/resources/prod/storage.tf:64]: you are right, thank you @gquadrati!

File: infra/resources/prod/storage.tf (hunk 3 above)
Comment (arcogabbo) [reply infra/resources/prod/storage.tf:52]: unfortunately, declaring at least 1 option obligates you to declare `change_feed` and `immutability_policy` objects following the current type definition
```hcl
  type = object({
    restore_policy_days   = optional(number, 0)
    delete_retention_days = optional(number, 0)
    last_access_time      = optional(bool, false)
    versioning            = optional(bool, false)
    change_feed = object({
      enabled           = optional(bool, false)
      retention_in_days = optional(number, 0)
    })
    immutability_policy = object({
      enabled                       = optional(bool, false)
      allow_protected_append_writes = optional(bool, false)
      period_since_creation_in_days = optional(number, 730)
    })
  })

```

File: infra/resources/prod/storage.tf (hunk 4 above)
Comment (arcogabbo) [reply infra/resources/prod/storage.tf]: done in https://github.com/pagopa/io-auth-n-identity-domain/pull/124/commits/c30ebf730ef51da4fd69ecc05107a4292d23abca

**Review by gquadrati — APPROVED** [review]

**Comment by arcogabbo** [comment]
closing because of https://github.com/pagopa/io-infra/pull/1313

**Pull request closed without merging**